js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -47,35 +53,42 @@ export function useIndexedDBCachedQuery<TQuery, TVariables extends OperationVari
         }
       }
     })();
-  }, [lru, version]);
-
-  const didFetch = React.useRef(false);
+  }, [lru, version, cacheBreaker]);
 
   const fetch = React.useCallback(async () => {
-    if (didFetch.current) {
-      return;
-    }
-    didFetch.current = true;
     setLoading(true);
+    if (fetchState[key]) {
+      return await new Promise<ApolloQueryResult<TQuery>>((res) => {
+        fetchState[key]?.onFetched.push((value) => {
+          setCacheBreaker((v) => v + 1);
+          res(value);
+        });
+      });
+    }
+    fetchState[key] = {onFetched: []};
     // Use client.query here so that we initially use the apollo cache if any data is available in it
     // and so that we don't subscribe to any updates to that cache (useLazyQuery and useQuery would both subscribe to updates to the
     // cache which can be very slow)
-    const {data} = await client.query<TQuery, TVariables>({
+    const queryResult = await client.query<TQuery, TVariables>({
       query,
       variables,
       fetchPolicy: 'no-cache', // Don't store the result in the cache,
       // should help avoid page stuttering due to granular updates to the data
     });
+    const {data} = queryResult;
     setLoading(false);
     lru.set(
       'cache',
       {data, version},
       {
-        expiry: new Date(Date.now() + ONE_WEEK),
+        expiry: new Date('3000'), // never expire,
```

Comment:
What does this do? What is the date 3000? It might be nice to say `3000-01-01` if this is interpreted as a year